snuba/datasets/entities/sessions.py

```diff
@@ -211,8 +212,24 @@ def __init__(self) -> None:
     def select_storage(
         self, query: Query, request_settings: RequestSettings
     ) -> StorageAndMappers:
-        granularity = extract_granularity_from_query(query, "started") or 3600
-        use_materialized_storage = granularity >= 3600 and (granularity % 3600) == 0
+        use_materialized_storage: Optional[bool] = None
+
+        # If the passed in `request_settings` arg is an instance of `SubscriptionRequestSettings`,
+        # then it is a crash rate alert subscription, and hence we decide on whether to use the
+        # materialized storage or the raw storage by examining the time_window.
+        # If the `time_window` <=1h, then select the raw storage otherwise select materialized
+        # storage
+        # NOTE: If we were to support other types of subscriptions over the sessions dataset that
+        # do not follow this method used to identify which storage to use, we would need to
+        # find a different way to distinguish them.
+        if isinstance(request_settings, SubscriptionRequestSettings):
+            from_date, to_date = get_time_range(query, "started")
+            if from_date and to_date:
+                use_materialized_storage = to_date - from_date > timedelta(hours=1)
```

Comment:
Yes, and we are okay with that .. its a tradeoff decision made for in favor of running the subscription logic every minute

Reply:
Ok, that sounds fine. I guess I'm confused why it's still possible to fall back to the non materialised storage later if either `to_date` or `from_date` is missing here then.

Reply:
It is not possible in subscription queries because there is no group by time column so if it doesn't actually go into line 228, it will always use the materialised view